src/Quota_Command.php

```diff
@@ -0,0 +1,230 @@
+			'where'  => $where,
+			'append' => $append,
+		];
+
+		$iterator = new TableIterator( $iterator_args );
+
+		$iterator = Utils\iterator_map(
+			$iterator,
+			function ( $blog ) {
+				return $this->determine_quotas( $blog );
+			}
+		);
+
+		if ( ! empty( $assoc_args['format'] ) && 'ids' === $assoc_args['format'] ) {
+			$sites     = iterator_to_array( $iterator );
+			$ids       = wp_list_pluck( $sites, 'blog_id' );
+			$formatter = new Formatter( $assoc_args, null, 'site' );
+			$formatter->display_items( $ids );
+		} else {
+			$formatter = new Formatter( $assoc_args, null, 'site' );
+			$formatter->display_items( $iterator );
+		}
+	}
+
+	/**
+	 * Determine quotas
+	 *
+	 * @param $blog
+	 *
+	 * @return mixed
+	 */
+	private function determine_quotas( $blog ) {
+		$blog->url = trailingslashit( get_site_url( $blog->blog_id ) );
+
+		switch_to_blog( $blog->blog_id );
+
+		// Get quota
+		$quota       = get_space_allowed(); // * MB_IN_BYTES  ???;
+		$blog->quota = $quota;
+
+		// Get quota used & quota used in percent
+		$used = get_space_used(); // * MB_IN_BYTES ???;
+
+		if ( $used > $quota ) {
+			$percentused = '100';
+		} else {
+			$percentused = ( $used / $quota ) * 100;
+		}
+		$blog->quota_used         = round( $used, 2 );
+		$blog->quota_used_percent = round( $percentused, 2 );
+
+		restore_current_blog();
+
+		return $blog;
+	}
+
+
+	/**
+	 * Returns the quota information for a single site
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<id>]
+	 * : The id of the site to get.
+	 *
+	 * [--fields=<fields>]
+	 * : Comma-separated list of fields to show.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - count
+	 *   - ids
+	 *   - json
+	 *   - yaml
+	 * ---
+	 *
+	 * ## AVAILABLE FIELDS
+	 *
+	 * These fields will be displayed by default for each site:
+	 *
+	 * * blog_id
+	 * * url
+	 * * quota
+	 * * quota_used
+	 * * quota_used_percent
+	 *
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Output the quota of blog 3
+	 *     $ wp quota get 3
+	 *     +---------+--------------------------------------------------------+--------------+----------------+---------------------+
+	 *     | blog_id | url                                                    | quota        | quota_used     | quota_used_percent  |
+	 *     +---------+--------------------------------------------------------+--------------+----------------+---------------------+
+	 *     | 3       | https://dev-site.local/subsite3                        | 10000        | 9850           | 98.50               |
+	 *     +---------+--------------------------------------------------------+--------------+----------------+---------------------+
+	 *
+	 * @subcommand get
+	 */
+	public function get( $args, $assoc_args ) {
+		if ( ! is_multisite() ) {
+			WP_CLI::error( 'This is not a multisite installation.' );
+		}
+
+		if ( empty( $args ) ) {
+			WP_CLI::error( 'Need to specify a blog id.' );
```

Comment:
Wie heute per Mail geschrieben. In Abhängigkeit von der Anzahl der Args (hier 0 oder 1) das Handling anpassen.
```
Pseudocode
1 == count( $args ) => Quota für Seite mit ID {args[0]} ausgeben
```
```
Pseudocode
empty( $args ) => Quota für aktuelle Seite ausgeben
```

Reply:
Ich habe es eben einmal mit einer Subdomain- und einer Subdirectory-Installation getestet:
Subdomain:
wp quota get --url=sub1.multisite-sub.local
+---------+----------------------------------+-------+------------+--------------------+
| blog_id | url                              | quota | quota_used | quota_used_percent |
+---------+----------------------------------+-------+------------+--------------------+
| 2       | http://sub1.multisite-sub.local/ | 100   | 0          | 0                  |
+---------+----------------------------------+-------+------------+--------------------+

wp quota get --url=sub1**irgendwas**.multisite-sub.local
Error: Site 'sub1irgendwas.multisite-sub.local' not found. Verify `--url=<url>` matches an existing site.

Soweit in Ordnung!

Subdirectory:
wp quota get --url=multisite.local/en
+---------+----------------------------+-------+------------+--------------------+
| blog_id | url                        | quota | quota_used | quota_used_percent |
+---------+----------------------------+-------+------------+--------------------+
| 2       | http://multisite.local/en/ | 200   | 1.65       | 0.82               |
+---------+----------------------------+-------+------------+--------------------+

wp quota get --url=multisite.local/en**IRGENDWAS**
+---------+-------------------------+-------+------------+--------------------+
| blog_id | url                     | quota | quota_used | quota_used_percent |
+---------+-------------------------+-------+------------+--------------------+
| 1       | http://multisite.local/ | 100   | 0          | 0                  |
+---------+-------------------------+-------+------------+--------------------+

wp quota get --url=multisite.localI**RGENDWAS**/en
Error: Site 'multisite.localirgendwas/en' not found. Verify `--url=<url>` matches an existing site.

Der --url-Parameter prüft offensichtlich nur den (sub)Domainnamen, aber nicht den Subdir-Namen.
Hinsichtlich auf "quota set" sehe ich da ein Problem, dass bei falscher Eingabe der SubDir die Hauptdomain gesetzt und dies womöglich nicht mal bemerkt wird. (weil die Message auch nicht immer gelesen wird...)

Reply:
Damit kann ich aktuell leben, da wir nur mit Subdomains arbeiten :-)

Reply:
Leider geht das "--url="-Thema doch noch weiter. Beim "list"-Kommando habe ich keine "$args"-Parameter. Da also drauf zu prüfen, ob gesetzt oder nicht ist quatsch. So lange ich also den globalen "--url="-Parameter nicht prüfen kann, kann ich das äquivalent zu "wp quota get <id>" nicht einbauen.

Reply:
Dann macht `list` eben nur die Auflistung für alle Seiten :-) Auch kein Problem. `get` gibt dann einzelne Datensätze zurück. Das passt schon .

Reply:
Ich hab die README im main branch gerade dahingehend angepasst, dass --url bei `list` nun nicht mehr relevant ist.